strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetArtifactController.java

```diff
@@ -681,26 +656,26 @@ private URI storePackage(String storageId,
 
             nugetArtifactManagementService.validateAndStore(storageId, repositoryId, path, nupkgFile.getStream());
 
-            File nuspecFile = File.createTempFile(nupkgFile.getId(), "nuspec");
-            try (FileOutputStream fileOutputStream = new FileOutputStream(nuspecFile))
+            Path nuspecFile = Files.createTempFile(nupkgFile.getId(), "nuspec");
+            try (OutputStream outputStream = Files.newOutputStream(nuspecFile))
             {
-                nupkgFile.getNuspecFile().saveTo(fileOutputStream);
+                nupkgFile.getNuspecFile().saveTo(outputStream);
             }
             path = String.format("%s/%s/%s.nuspec", nupkgFile.getId(), nupkgFile.getVersion(), nupkgFile.getId());
 
-            nugetArtifactManagementService.validateAndStore(storageId, repositoryId, path, new FileInputStream(nuspecFile));
+            nugetArtifactManagementService.validateAndStore(storageId, repositoryId, path, Files.newInputStream(nuspecFile));
 
-            File hashFile = File.createTempFile(String.format("%s.%s", nupkgFile.getId(), nupkgFile.getVersion()),
+            Path hashFile = Files.createTempFile(String.format("%s.%s", nupkgFile.getId(), nupkgFile.getVersion()),
                                                 "nupkg.sha512");
-            nupkgFile.getHash().saveTo(hashFile);
+            nupkgFile.getHash().saveTo(Files.newOutputStream(hashFile));
 
             path = String.format("%s/%s/%s.%s.nupkg.sha512",
                                  nupkgFile.getId(),
                                  nupkgFile.getVersion(),
                                  nupkgFile.getId(),
                                  nupkgFile.getVersion());
 
-            nugetArtifactManagementService.validateAndStore(storageId, repositoryId, path, new FileInputStream(hashFile));
+            nugetArtifactManagementService.validateAndStore(storageId, repositoryId, path, Files.newInputStream(hashFile));
```

Comment:
@fuss86 many thanks for this too :smiley:

Reply:
+1

Reply:
:)